### PR TITLE
Users

### PR DESCRIPTION
**Pull request by mimansha-swarup**
- follow user
- unfollow user
- - unfollower modal
- - follower modal
-  update profile

### CODE REVIEW

File: src/component/Profile/EditModal.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                src={
+                  profilePictureData.file
+                    ? URL.createObjectURL(profilePictureData.file)
+                    : currUser?.profilePicture
+                }
+                sx={{ height: 156, width: 156 }}
+              />
+              <input
+                accept="image/*"
+                id="profilePicture"
```
Comment (Sen-442b) [inline src/component/Profile/EditModal.js]: you can follow  kebab casing here

File: src/component/Profile/EditModal.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                name="profilePicture"
+                style={{ display: "none" }}
```
Comment (Sen-442b) [inline src/component/Profile/EditModal.js:109]: avoid inline styling if possible

File: src/component/Profile/ProfileHeader.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  const selectUser = getCurrentUser(users, userData.username);
+
+  const handleActionChange = (actionName) =>
+    setActions((prevAction) => ({
+      ...prevAction,
+      [actionName]: !prevAction[actionName],
+    }));
+
+  const handleFollowUser = () =>
+    dispatch(followUser({ followerId: currUser._id, token }));
```
Comment (Sen-442b) [inline src/component/Profile/ProfileHeader.js:44]: you can destructure `currUser` prop at the top of   your component

File: src/server.js [hunk 4]
```diff
@@ -116,6 +116,7 @@ export function makeServer({ environment = "development" } = {}) {
         "/users/unfollow/:followUserId/",
         unfollowUserHandler.bind(this)
       );
+      this.passthrough("https://api.cloudinary.com/v1_1/daz8z0ksy/image/upload")
```
Comment (Sen-442b) [inline src/server.js:119]: you can replace the query string with `**` after the namespace  if you want to use different assets from the same domain 
`this.passthrough("https://api.cloudinary.com/v1_1/**")`

**Review by Sen-442b — APPROVED** [review]
-  Code looks clean 
-  proper folder structure 
- UI looks clean as well

File: src/component/Profile/EditModal.js (hunk 2 above)
Comment (mimansha-swarup) [reply src/component/Profile/EditModal.js:109]: i will keep that in mind

File: src/server.js (hunk 4 above)
Comment (mimansha-swarup) [reply src/server.js:119]: thanks for telling this anyway i don't need now
will change if needed

File: src/component/Home/CurrentAvatar.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import React from "react";
+import { useSelector } from "react-redux";
+import { getCurrentUser } from "../../utils";
+
+export const CurrentAvatar = ({ sx = {} }) => {
+  const {
+    auth: { userData },
+    user: { users },
+  } = useSelector((state) => state);
+  let currUser = getCurrentUser(users, userData.username);
```
Comment (manojasarsa) [inline src/component/Home/CurrentAvatar.js:11]: You can make the variable const to be consistent throughout the code.

File: src/component/Home/CurrentAvatar.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  currUser = currUser? currUser : userData
+  
+  return <Avatar src={currUser.profilePicture} alt="ProfilePicture" sx={sx} />;
```
Comment (manojasarsa) [inline src/component/Home/CurrentAvatar.js:14]: Good way of creating a separate component for Avatar.

File: src/component/Profile/EditModal.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+          method: "POST",
+          body: formData,
+        });
+        console.log(response);
+        const { url } = await response.json();
+        console.log(url);
+        setProfilePictureData((prev) => ({ ...prev, url }));
+        setProfilePictureData((prev) => ({ ...prev, loading: false }));
+      } catch (err) {
+        console.log(err);
```
Comment (manojasarsa) [inline src/component/Profile/EditModal.js]: you can use `console.error` here

File: src/component/Profile/EditModal.js [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+      website: event.target.website.value
+        ? event.target.website.value
+        : currUser.website,
+
+      profilePicture: profilePictureData.url
+        ? profilePictureData.url
+        : currUser.profilePicture,
+    };
+    console.log(profilePictureData);
```
Comment (manojasarsa) [inline src/component/Profile/EditModal.js]: you can remove these console.log if no longer needed

File: src/component/Profile/EditModal.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+                  }));
+                }}
+              />
+
+              <label style={UploadImageStyle} htmlFor="profile-Picture">
+                <AddAPhoto style={UploadIconStyle} />
+              </label>
+            </Box>
+            <Typography variant="body1" component="p">
+              <strong>Username :</strong> @{`${currUser.username}`}
```
Comment (manojasarsa) [inline src/component/Profile/EditModal.js:125]: Nice use of `strong` tag to highlight the username

File: src/features/user/userSlice.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        if (user.username === payload.user.username) return payload.user;
+        return user;
+      });
+    },
+    [unfollowUser.pending]: (state, { payload }) => {
+      console.log("pending",payload)
+      
+    },
+    [unfollowUser.reject]: (state, { payload }) => {
+      console.log("reject",payload)
```
Comment (manojasarsa) [inline src/features/user/userSlice.js:35]: You can remove these two states if you are not doing any operation like loader or showing errors. Else, you can put a TODO above it if you are planning to do it later.

**Review by manojasarsa — APPROVED** [review]
The UI is looking clean and neat.
The code seems pretty structured. 
Mentioned some pointers which you can look into.

File: src/component/Home/CurrentAvatar.js (hunk 5 above)
Comment (mimansha-swarup) [reply src/component/Home/CurrentAvatar.js:11]: on next line its being reassigned

**Pull request merged**